account_invoice_supplier_self_invoice/models/res_partner.py

```diff
@@ -0,0 +1,46 @@
+# © 2017 Creu Blanca
+# License AGPL-3.0 or later (https://www.gnuorg/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    self_invoice = fields.Boolean(
+        string="Approves self invoice",
+        default=False,
+        help="When checked, all invoices will generate a self-invoice " "on validation",
+    )
+    self_invoice_sequence_id = fields.Many2one(
+        comodel_name="ir.sequence", string="Self invoice sequence", ondelete="restrict"
+    )
+    self_invoice_report_footer = fields.Text(
+        string="Self invoice footer",
+        help="Footer text displayed at the bottom of the self invoice reports.",
+    )
```

Comment:
Shouldn't all these fields be company-dependant?

Reply:
Not really...

If we keep several sequences according to each company, we need to show different numbers and that might be hard to manage (the creation of the sequence might be a problem....). IMO it is easier to make it only once to simplifie the process.

Reply:
So let's say I have companies A and B, and I have supplier C which will self-invoice me.

C would have the same self-invoice sequence for A and B? I don't think so.

What if C self-invoices only B? Odoo would be suggesting self-invoices for A purchase orders. Not good IMHO.

Reply:
I think Yajo is in the correct position.

Sequence has the `company_id` field, so the creation I think will not be a problem.

Reply:
Great point @yajo 

I think this is obvious, all the sequences should be by res.company . Also in de case:

- Company A
- Company B
- Vendor Self invoice 1

Case 1:

- Company A make self-biiling to "Vendor Self invoice 1" so invoice is SI/INV/2022/11/0001
- Company B make self-biiling to "Vendor Self invoice 1" so invoice is SI/INV/2022/11/0001, absolutely other invoice sequence. Company A & B are different legal entities.

Case 2: 

- Company A make self-biiling to "Vendor Self invoice 1" so invoice is SI/INV/2022/11/0001
- Company B DON'T make self-biiling to "Vendor Self invoice 1" so invoice is the number the vendor send us.


Any other use case is not correct. Multi-company resilience please!

Thank you!

Reply:
This is broking previous installations of this module, also, both systems have sense, so, IMO, or another glue module is needed to make it company_dependent or find a way to keep the current behavior please.

I have been working with this approach for 3 years with Odoo and 20 with other kind of softwares, and it is a valid options.